pl_bolts/models/self_supervised/cpc/cpc_module.py

```diff
@@ -33,65 +33,52 @@ class CPCV2(pl.LightningModule):
 
     def __init__(
             self,
-            datamodule: Optional[pl.LightningDataModule] = None,
             encoder_name: str = 'cpc_encoder',
             patch_size: int = 8,
             patch_overlap: int = 4,
-            online_ft: int = True,
+            online_ft: bool = True,
             task: str = 'cpc',
             num_workers: int = 4,
-            learning_rate: int = 1e-4,
-            data_dir: str = '',
-            batch_size: int = 32,
+            num_classes: int = 10,
+            learning_rate: float = 1e-4,
             pretrained: Optional[str] = None,
             **kwargs,
     ):
         """
         Args:
-            datamodule: A Datamodule (optional). Otherwise set the dataloaders directly
             encoder_name: A string for any of the resnets in torchvision, or the original CPC encoder,
                 or a custon nn.Module encoder
             patch_size: How big to make the image patches
-            patch_overlap: How much overlap should each patch have.
-            online_ft: Enable a 1024-unit MLP to fine-tune online
+            patch_overlap: How much overlap each patch should have
+            online_ft: If True, enables a 1024-unit MLP to fine-tune online
             task: Which self-supervised task to use ('cpc', 'amdim', etc...)
-            num_workers: num dataloader worksers
-            learning_rate: what learning rate to use
-            data_dir: where to store data
-            batch_size: batch size
+            num_workers: number of dataloader workers
+            num_classes: number of classes
+            learning_rate: learning rate
             pretrained: If true, will use the weights pretrained (using CPC) on Imagenet
         """
 
         super().__init__()
         self.save_hyperparameters()
 
-        # HACK - datamodule not pickleable so we remove it from hparams.
-        # TODO - remove datamodule from init. data should be decoupled from models.
-        del self.hparams['datamodule']
-
         self.online_evaluator = self.hparams.online_ft
 
         if pretrained:
             self.hparams.dataset = pretrained
             self.online_evaluator = True
 
-        assert datamodule
-        self.datamodule = datamodule
-
         self.encoder = self.init_encoder()
 
         # info nce loss
         c, h = self.__compute_final_nb_c(self.hparams.patch_size)
         self.contrastive_task = CPCTask(num_input_channels=c, target_dim=64, embed_scale=0.1)
 
         self.z_dim = c * h * h
-        self.num_classes = self.datamodule.num_classes
+        self.num_classes = num_classes
```

Comment:
Since `self.num_classes` was already defined, I'm leaving it as is, but is this variable really necessary? As far as I understand [the paper](https://arxiv.org/abs/1905.09272), it uses only images without any labels...

Reply:
Is this variable `self.num_classes` useful/necessary for downstream tasks...?